odoo/fields.py

```diff
@@ -737,6 +725,10 @@ def _description_help(self, env):
     # Conversion of values
     #
 
+    def cache_key(self, env):
+        """ Return the extra key in cache to get/set a value. """
+        return env if self.context_dependent else (env.cr, env.uid)
```

Comment:
Wouldn't be nicer to pass a record instead of an environment here?

This may open the possibilities to fix further issues, like records with a `NewId`. The current issue with those records is: when you switch to a different environment, your record has lost its data, which is present only in its first environment's cache. We could make those records have essentially a single cache.

I am open to suggestions.